pkg/sources/reconciler/cloudeventssource/adapter.go

```diff
@@ -99,16 +96,34 @@ func (r *Reconciler) BuildAdapter(src commonv1alpha1.Reconcilable, sinkURI *apis
 			}
 		}
 
-		if len(kvs) != 0 {
+		if len(kvs) > 0 {
 			s, err := json.Marshal(kvs)
 			if err != nil {
 				return nil, fmt.Errorf("serializing keyMountedValues to JSON: %w", err)
 			}
-			options = append(options, resource.EnvVar(envCloudEventsBasicAuthCredentials, string(s)))
+
+			authEnvs = append(authEnvs, corev1.EnvVar{
+				Name:  envCloudEventsBasicAuthCredentials,
+				Value: string(s),
+			})
 		}
 	}
 
-	return common.NewAdapterKnService(src, sinkURI, options...), nil
+	ceOverridesStr := cloudevents.OverridesJSON(typedSrc.Spec.CloudEventOverrides)
+
+	return common.NewAdapterKnService(src, sinkURI,
+		resource.Image(r.adapterCfg.Image),
+
+		resource.VisibilityPublic,
+
+		resource.Volumes(authVolumes...),
+		resource.VolumeMounts(authVolumeMounts...),
+		resource.EnvVars(authEnvs...),
```

Comment:
One of the nice properties of `append()` is that the operation is a no-op if the argument is empty.
So if auth isn't set, these 3 actions won't fail but the constructor remains easy to read (I think).